addition of parent Auth class

amanigaultw opened the pull request:
**Please describe the bug this fixes or the feature this adds.**
a parent (i.e., Auth) class was added to streamline the process of updating methods shared by the Credentials and Token classes.

**Please describe how you tested this change. Include unit tests whenever possible.**
pynssp Pipeline tests were executed successfully

**Did you create or modify any associated documentation with this change? If documentation is not included in PR, please link to related documentation.**
NA

**If you added or modified HTML, did you check that it was 508 compliant?**
NA

**Please tag any specific reviewers you would like to review this PR**
Roseric Azondekon

**Please include the following checks for open source contributing?**

* [x] Did you check for sensitive data, and remove any?
* [x] Are additional approvals needed for this change?
* [x] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

amanigaultw committed:
added Auth parent class

Added a new abstract class called Auth from which methods shared by credentials and token can be managed.

amanigaultw committed:
Update check-standard.yaml

added dev branch to check-standard

amanigaultw committed:
updated documentation

reintroduced documentation that was deleted during refactoring

amanigaultw committed:
Merge branch 'dev' of https://github.com/amanigaultw/pynssp into dev

amanigaultw committed:
Update check-standard.yaml

rosericazondekon reviewed (changes requested):
Hey @amanigaultw,
Thank you for your contribution. Please, provide a docstring to the `Auth` class. You may refer to the `Credentials` or `Token` classes for guidance.

amanigaultw committed:
Addressed PR feedback

- removed filename from init args
- removed ABC dependency
- added docstring for Auth class

amanigaultw commented:
Thank you for your feedback!

The latest commit addressed the requested change, such that a docstring was added for Auth class; this docstring does not contain parameters or examples as neither would be appropriate for this abstract class.  In addition, the ABC package was not necessary to implement desired changes and thus the dependency were removed.  Lastly, the filename argument was removed from the object initialization method (for Auth, Token, and Credentials classes), as user are not expected to set this variable.  

Please advise if additional changes are required.
